assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -0,0 +1,170 @@
+import {
+	MODULES_ANALYTICS_4,
+	PROPERTY_CREATE,
+	WEBDATASTREAM_CREATE,
+} from '../../../analytics-4/datastore/constants';
+import { FORM_SETUP, MODULES_ANALYTICS } from '../../datastore/constants';
+import { Select, Option } from '../../../../material-components';
+import { GA4ActivateSwitch } from '../common';
+import {
+	PropertySelect,
+	UseSnippetSwitch,
+} from '../../../analytics-4/components/common';
+const { useSelect, useDispatch } = Data;
+
+export default function GA4SettingsControls() {
+	const [ matchedProperty, setMatchedProperty ] = useState();
+	const [ matchedWebDataStream, setMatchedWebDataStream ] = useState();
+
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	// This select is needed to check whether the AdminAPI works or not.
+	useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getProperties( accountID )
+	);
+
+	const isAdminAPIWorking = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isAdminAPIWorking()
+	);
+
+	const enableGA4 = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'enableGA4' )
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const {
+		matchAccountProperty,
+		matchWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	} = useDispatch( MODULES_ANALYTICS_4 );
+
+	useUpdateEffect( () => {
+		const matchGA4Information = async () => {
+			const matchingProperty = await matchAccountProperty( accountID );
+
+			setMatchedProperty( matchingProperty );
+			if ( matchingProperty?._id ) {
+				const matchingWebDataStream = await matchWebDataStream(
+					matchingProperty._id
+				);
+				setMatchedWebDataStream( matchingWebDataStream );
+			}
+		};
+
+		if ( isAdminAPIWorking ) {
+			matchGA4Information();
+		}
+	}, [
+		accountID,
+		matchAccountProperty,
+		matchWebDataStream,
+		isAdminAPIWorking,
+	] );
+
+	const onActivate = useCallback( () => {
+		setPropertyID( matchedProperty?._id || PROPERTY_CREATE );
+		setWebDataStreamID( matchedWebDataStream?._id || WEBDATASTREAM_CREATE );
+		setMeasurementID( matchedWebDataStream?.measurementId || '' ); // eslint-disable-line sitekit/acronym-case
+	}, [
+		matchedProperty,
+		matchedWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	] );
+
+	if ( ! isAdminAPIWorking ) {
+		return null;
+	}
+
+	const isDisabled = ! propertyID && ! enableGA4;
+
+	return (
+		<div className="googlesitekit-settings-module__fields-group">
+			<div className="googlesitekit-setup-module__inputs">
+				{ ! isDisabled && (
+					<PropertySelect
+						label={ __(
+							'Google Analytics 4 Property',
+							'google-site-kit'
+						) }
+					/>
+				) }
+				{ isDisabled && (
+					<Select
+						className="googlesitekit-analytics__select-property"
+						label={ __(
+							'Google Analytics 4 Property',
+							'google-site-kit'
+						) }
+						value={ matchedProperty?._id || '' }
```

Comment:
The value isn't really important here, it should always be "selected" though because otherwise the floating label moves into the input area and is cut off.
```suggestion
						value="1"
```

Reply:
@aaemnnosttv see this comment: https://github.com/google/site-kit-wp/pull/4229#discussion_r739488882

> It should instead show its label within the dropdown field, which is the default behavior if there is no selection being made - that's why value needs to be empty.

Reply:
@aaemnnosttv Yes the label should be within the select when it's empty, this is the expected "empty" behavior of all our selects. So I think this is correct as is.

Reply:
Yes, but for this specific select it looks broken because the text doesn't fit.
![image](https://user-images.githubusercontent.com/1621608/140191970-f79a379e-6b2c-4135-b39c-d5b999877264.png)

Reply:
@felixarntz this isn't particularly blocking from me but IMO it looks better and is a bit more consistent with the rest of the interface when the label is over the input here.

Reply:
@aaemnnosttv Right, but then a better fix would be to extend the width of the dropdown. A very long property ID for example would also be cut off like that, so I don't think we should fix it by enforcing the label to appear above, which would be inconsistent with the intended design.

Reply:
> A very long property ID for example would also be cut off like that

Actually it does not cut off the actual value in the select, it will stretch to show the whole thing. So the truncated text seems to only apply to the label. More text can fit when it is above due to the smaller size though so this is less likely to happen.
![image](https://user-images.githubusercontent.com/1621608/140277647-0ed867cf-b3a4-4c85-a6fa-e19c056a6d0c.png)